main/src/cgeo/geocaching/utils/FileUtils.java

```diff
@@ -96,6 +106,232 @@ public static boolean deleteDirectory(@NonNull final File dir) {
         return delete(dir);
     }
 
+    /**
+     * Moves a file/directory to a new name. Tries a rename first (faster) and falls back
+     * to a copy + delete. Target directories are created if needed.
+     *
+     * @param source
+     * @param target
+     * @return true, if successfully
+     */
+    public static boolean move(final File source, final File target) {
+        if (!source.exists() || target.exists()) {
+            return false;
+        }
+        FileUtils.mkdirs(target.getParentFile());
+        boolean success = source.renameTo(target);
+        if (!success) {
+            // renameTo might fail across mount points, try copy/delete instead
+            success = FileUtils.copy(source, target);
+            if (success) {
+                FileUtils.deleteDirectory(source);
+            } else {
+                Log.w("Couldn't move " + source + " to " + target);
+            }
+        }
+        return success;
+    }
+
+    /**
+     * Moves a file/directory into the targetDirectory.
+     *
+     * @param source
+     * @param targetDirectory
+     */
+    public static void moveTo(final File source, final File targetDirectory) {
+        move(source, new File(targetDirectory, source.getName()));
+    }
+
+    /**
+     * Get the guessed file extension of an URL. A file extension can contain up-to 4 characters in addition to the dot.
+     *
+     * @param url
+     *            the relative or absolute URL
+     * @return the file extension, including the leading dot, or the empty string if none could be determined
+     */
+    @NonNull
+    public static String getExtension(@NonNull final String url) {
+        final String urlExt;
+        if (url.startsWith("data:")) {
+            // "data:image/png;base64,i53…" -> ".png"
+            urlExt = StringUtils.substringAfter(StringUtils.substringBefore(url, ";"), "/");
```

Comment:
You have to handle `null` if you use `substringBetween` (the current construct returns an empty string) if it doesn't match. I'm not sure the subsequent call to `defaultString` will get better performances, but it may indeed may more readable as

    StringUtils.defaultString(StringUtils.substringBetween("/", ";"));